editions/tw5.com/tiddlers/pragmas/Pragma_ _procedure.tid

```diff
@@ -52,4 +52,59 @@ Procedure definitions can be nested by specifying the name of the procedure in t
 \end special-button
 
 <<special-button>>
+""">>
+
+! Use of Parameters Inside Procedures
+The parameters can be declared inside procedures. The parameters widget is necessary in a procedure if you want to use computed default values. For example:
+
+<<wikitext-example-without-html 
+src:"""\procedure myproc()
+<$parameters name={{$:/SiteTitle}} desc={{$:/SiteSubtitle}}>
+This is <<name>> demonstrates <<desc>>.
+</$parameters>
+\end
+
+<<myproc>>
+""">>
+
+!! Caution in Using Positional Parameters
+As procedure is a shortcut syntax for the SetVariableWidget with an implicit ParametersWidget, there doesn't seem to be a good reason to have multiple parameters widgets within a definition. In the below example when passing `x` to `myproc`, it will also be set to `a`:
```

Comment:
We try to avoid contractions like "doesn't". Might be better to say:

Procedures are a shortcut syntax for the SetVariableWidget with an implicit ParametersWidget, so generally there is no reason to have multiple parameters widgets within a definition